### DISCUSSION TITLE
How can I load morph animations using Assimp?

### DISCUSSION THREAD

**Discussion by suikameronn** [assimp/assimp#5744] [Q&A] [answered]
I'm having trouble loading morph animations using Assimp. I've been looking into the aiAnimation structure and noticed the mNumMorphMeshChannels member, but I'm unsure how to use it to extract and animate the morph data.

Could someone please point me in the right direction? Any help would be greatly appreciated. 

![VulkanProject - Microsoft Visual Studio 2024_08_31 20_40_58](https://github.com/user-attachments/assets/fb20febd-c20c-4c61-8a5a-a76af1642bd8)
409-8f62-7d82a6ecbb43)

**Comment by kimkulling** [comment, marked as answer]
I haven't used them right now. However, the common concept behind it is the same as the concept of skinning. YOu have a list of affected vertices in your mesh. In the update you just neet to check which vertices are used, interpolate them for the given time, and update the position and their texturing in the shader.
I am currently improving the new assimp viewer for animation, because most of the animation stuff was not written by me. Hopefully, I will be able to help you more in the next months.
